agent/api/task.go

```diff
@@ -202,23 +208,64 @@ func (task *Task) initializeCredentialsEndpoint(credentialsManager credentials.M
 
 }
 
+// BuildCNIConfig constructs the cni configuration from eni
+func (task *Task) BuildCNIConfig() (*ecscni.Config, error) {
+	if !task.isNetworkModeVPC() {
+		return nil, errors.New("task isnot enabled with vpc networkmode")
+	}
+
+	cfg := &ecscni.Config{}
+	eni := task.GetTaskENI()
+	if eni == nil {
```

Comment:
I meant that you don't need to check this again here. You can assume that it won't return `nil`